lib/foreman_templates/engine.rb

```diff
@@ -16,6 +16,17 @@ class Engine < ::Rails::Engine
     initializer 'foreman_templates.register_plugin', :before => :finisher_hook do
       Foreman::Plugin.register :foreman_templates do
         requires_foreman '>= 1.12'
+
+        security_block :templates do
+          permission :import_templates, {
+            :template => [:import],
+            :"api/v2/template" => [:import]
+          }, :resource_type => 'Template' 
+        end
+
+        role = Role.where(:name => 'Manager').first
+        role.add_permissions!('import_templates') unless role.permissions.find_by_name('import_templates')
```

Comment:
There's a DSL for defining permissions, see [this](https://github.com/theforeman/foreman_openscap/blob/master/lib/foreman_openscap/engine.rb#L53) for example. Roles are defined like [this](https://github.com/theforeman/foreman_openscap/blob/master/lib/foreman_openscap/engine.rb#L109) but @xprazak2 works on better API for plugins in https://github.com/theforeman/foreman/pull/4168 (maybe your input as plugin developer would move the discussion on foreman-dev mailing list further)